_posts/2022-08-05-crossplane-demo.md

```diff
@@ -0,0 +1,231 @@
+
+```yaml
+  providerConfigRef:
+    name: provider-aws
+```
+
+The last section references which `ProviderConfig` to use. Of particular note, this means you could have a single `Provider` (such as AWS) and several `ProviderConfig` objects that each hit different accounts or have different permissions - this allows you to still have one place with all the credentials at that the author of the yaml doesn't need to see or know, they just need to know the name of the `ProviderConfig`.
+
+Okay, so you take this yaml, tweak it to your settings and then throw it at the cluster
+
+```bash
+kubectl apply -f s3_site.yaml
+```
+
+And then what? This is where Crossplane does its magic - it reaches out to `Provider` with the `ProviderConfig` and creates the object and grabs the information about it. 
+
+Neat, huh?
+
+Well, let's try something bigger - an Azure Kuberenetes cluster.
+
+_Note: At the time of writing this, non-corporate accounts have a hard time getting resources in the US regions, thus why I chose Norway for the example._
+
+Looking through my [example](https://gitlab.com/mjh/crossplane-demo/-/blob/main/crossplane/norway_aks.yaml), we can see that creating an AKS isn't just one object, but four of them and they're all in the same yaml file:
+
+  * ResourceGroup
+  * VirtualNetwork
+  * Subnet
+  * AKSCluster
+
+And more importantly, we can see that they reference one another, such as `resourceGroupNameRef` referencing the first object created. This isn't tied to just one yaml file; in fact, this could be 4 yaml files and they could reference each other's object. This was something I was concerned about in coming from Terraform; I do a lot of references and glad to see they're still here.
+
+So, once this `norway_aks.yaml` is ran, what happens? Quite frankly, we wait at first, Kubernetes wasn't built in a day. But after about 5 minutes, we can take a quick look for sure!
+
+!["Using kubectl to describe the AKS Cluster"](/assets/images/posts/2022-08-05-crossplane/describe-akscluster.png)
+
+Alright, there's a lot of notes in here, but let's skip down to the relevant part of this object.
+
+!["AKS Status Successful"](/assets/images/posts/2022-08-05-crossplane/aks-status-state.png)
+
+So, we're cooking with gas now! You can even verify in the Azure console if you really want.
+
+!["Status in Azure Console of AKS"](/assets/images/posts/2022-08-05-crossplane/aks-in-azure-console.png)
+
+So, now that it's in state, what if I don't want it anymore?
+
+Delete the yaml file!
+
+As you can guess by the fact I have the DigitalOcean provider and configuration, I had some DigitalOcean resources. However, that Provider is still maturing and would created a brand new Droplet every time the status of configuration was checked - 8 droplets later, I hastily deleted the yaml file and applied it and the Droplets stopped being created. The DO Provider is still not good at deleting resources, but the AWS one is!
```

Comment:
Any issue replacing "DO" with "DigitalOcean"? You use the full name twice in the paragraph and then the abbreviation. The abbreviation wasn't used elsewhere except a token variable.

Reply:
Good call, corrected